From 8b8e4e725f02a9b26bdd2316dafd0dba694be098 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sat, 12 May 2018 00:33:52 +0200
Subject: [PATCH 10/11] lineage: Isolate LineageOS versions properties for
 build.prop

 * Our properties were supposed to go to /system/etc/prop.default
    after the following commit:
    "lineage: Move to Google's method of defining system default props"
    Change-Id: I6cb0e28a7599b010b389cc541015a37010a00f4b

 * However if BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED is not true,
    only /default.prop will retain the properties contents of
    ADDITIONAL_DEFAULT_PROPERTIES and PRODUCT_SYSTEM_DEFAULT_PROPERTIES,
    and none of our versioning identification was held in the system

 * Enabling BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED globally would
    break all properties on devices that handle partitions usually
    at the device level rather than the kernel due to mounting races

 * Create a new versions.mk file to isolate LineageOS properties
    that define the system's versions identifications, and
    use the ADDITIONAL_BUILD_PROPERTIES internal build variable
    to include it from build/make in the proper place

Change-Id: I0060141c097b3d14c3710eee1e0caf7110634967
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 config/common.mk             | 10 ----------
 config/lineage_sdk_common.mk |  9 ---------
 config/versions.mk           | 19 +++++++++++++++++++
 3 files changed, 19 insertions(+), 19 deletions(-)
 create mode 100644 config/versions.mk

diff --git a/config/common.mk b/config/common.mk
index 4044fd75..f2887d63 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -335,13 +335,6 @@ else
     endif
 endif
 
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    ro.lineage.version=$(LINEAGE_VERSION) \
-    ro.lineage.releasetype=$(LINEAGE_BUILDTYPE) \
-    ro.lineage.build.version=$(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR) \
-    ro.modversion=$(LINEAGE_VERSION) \
-    ro.lineagelegal.url=https://lineageos.org/legal
-
 PRODUCT_EXTRA_RECOVERY_KEYS += \
     vendor/lineage/build/target/product/security/lineage
 
@@ -372,9 +365,6 @@ ifneq ($(PRODUCT_DEFAULT_DEV_CERTIFICATE),build/target/product/security/testkey)
 endif
 endif
 
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    ro.lineage.display.version=$(LINEAGE_DISPLAY_VERSION)
-
 -include $(WORKSPACE)/build_env/image-auto-bits.mk
 -include vendor/lineage/config/partner_gms.mk
 
diff --git a/config/lineage_sdk_common.mk b/config/lineage_sdk_common.mk
index f1467c85..ab6629f3 100644
--- a/config/lineage_sdk_common.mk
+++ b/config/lineage_sdk_common.mk
@@ -39,12 +39,3 @@ ifndef LINEAGE_PLATFORM_REV
   # If you are doing a release and this is NOT 0, you are almost certainly doing it wrong
   LINEAGE_PLATFORM_REV := 0
 endif
-
-# LineageOS Platform SDK Version
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-  ro.lineage.build.version.plat.sdk=$(LINEAGE_PLATFORM_SDK_VERSION)
-
-# LineageOS Platform Internal
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-  ro.lineage.build.version.plat.rev=$(LINEAGE_PLATFORM_REV)
-
diff --git a/config/versions.mk b/config/versions.mk
new file mode 100644
index 00000000..e2d4062c
--- /dev/null
+++ b/config/versions.mk
@@ -0,0 +1,19 @@
+# LineageOS System Version
+ADDITIONAL_BUILD_PROPERTIES += \
+    ro.lineage.version=$(LINEAGE_VERSION) \
+    ro.lineage.releasetype=$(LINEAGE_BUILDTYPE) \
+    ro.lineage.build.version=$(PRODUCT_VERSION_MAJOR).$(PRODUCT_VERSION_MINOR) \
+    ro.modversion=$(LINEAGE_VERSION) \
+    ro.lineagelegal.url=https://lineageos.org/legal
+
+# LineageOS Platform Display Version
+ADDITIONAL_BUILD_PROPERTIES += \
+    ro.lineage.display.version=$(LINEAGE_DISPLAY_VERSION)
+
+# LineageOS Platform SDK Version
+ADDITIONAL_BUILD_PROPERTIES += \
+  ro.lineage.build.version.plat.sdk=$(LINEAGE_PLATFORM_SDK_VERSION)
+
+# LineageOS Platform Internal
+ADDITIONAL_BUILD_PROPERTIES += \
+  ro.lineage.build.version.plat.rev=$(LINEAGE_PLATFORM_REV)
-- 
2.17.0

